crates/integrations/datafusion/src/table/mod.rs

```diff
@@ -134,11 +152,18 @@ impl TableProvider for IcebergTableProvider {
             self.table.clone(),
             self.snapshot_id,
             self.schema.clone(),
+            self.statistics
+                .clone()
+                .unwrap_or(Statistics::new_unknown(self.schema.as_ref())),
```

Comment:
One major missing aspect is applying the filters here (and perhaps projections) to the base statistics to further reduce the range of upper-lower bounds.

Reply:
Marking this as Draft until i add support for this.